### ISSUE TITLE
new function for signal/rxjs interop

### ISSUE THREAD

**Issue by ymajoros** [angular/angular#57946]
### Description

I propose some function to run some computation that returns an Observable, and convert it to a signal.

Typical workflow:
- inputs = signals
- when they change, run some computation (typically, http requests)
- convert to a signal

### Proposed solution

```typescript
export function computedFromObservable<T>(computation: () => Observable<T>, options?: CreateComputedOptions<Observable<T>>): Signal<T | undefined> {
  const computedSignal = computed(computation, options);
  const computedObservable = toObservable(computedSignal);
  const observable = computedObservable.pipe(
    switchAll()
  );
  return toSignal(observable);
}
```

### Alternatives considered

Originally, I something like this, but it lacked the power of `computed()`:
```typescript
export function pipeSignalWithDefault<T, U>(signal: Signal<T>, pipe: OperatorFunction<T, U>, initialValue: U): Signal<U> {
  const signalAsObservable = toObservable(signal);
  const signalPipe = signalAsObservable.pipe(pipe);

  return toSignal(signalPipe, {initialValue});
}
```

**Comment by alxhub** [comment]
Alternatively, `toObservable` could be relaxed to take `() => T` instead of `Signal<T>` - there's no technical reason why it needs to be a single signal...

**Comment by ymajoros** [comment]
> Alternatively, `toObservable` could be relaxed to take `() => T` instead of `Signal<T>` - there's no technical reason why it needs to be a single signal...

Yes, but I'm not sure I understand how it solves the problem. I do like the computed() semantics, but it happens that the typical (e.g. http) internal result is an Observable.

**Comment by ymajoros** [comment]
I added some explanation about usage. Here is an example:
```typescript
    this.productPage = computedFromObservable(() => this.productService.find$(this.productSearch(), this.pagination()));
```

... replacing this more complicated construct:
```typescript
    const productParams: Signal<[ProductSearch, Pagination]> = computed(() => [this.preparationSearch(), this.pagination()]);
    const productPageObservable = toObservable(productParams).pipe(
      switchMap(([productSearch, pagination]) => this.productService.find$(productSearch, pagination))
    );
    this.productPage = toSignal(productPageObservable);
```

**Comment by Csellers15** [comment]
+1 on this

**Comment by alixroyere** [comment]
That would be nice to have a computedToSignal that could subscribe to an observable and return a signal of the values, depending on an other signal. As a mix between toSignal and computed.
Or any other short way to have backend call result value in a signal, depending on a component input signal

**Comment by Gallimathias** [comment]
Thank you for this suggestion, we have now adopted it in our company.

We very often have the case in a large application that we have observables due to other singals.

For me, this is definitely a necessary or interesting addition.

It corresponds somewhat to the defer that we already have for observables.

a deferToSignal, so to speak

**Comment by JeanMeche** [comment]
This request seems to be covereed by `rxResource` today.

**Comment by ymajoros** [comment]
> This request seems to be covereed by `rxResource` today.

Indeed, we moved on to use linkedSignal(), then even resource()/rxResource(), then even httpResource() when eager loading was acceptable.
